### ISSUE TITLE
set duration (instead of start+end)

### ISSUE THREAD

**Issue by wivaku** [jsGanttImproved/jsgantt-improved#191]
I have a project where I don't know (all the) start/end dates, but I do know the durations.

Is it possible to enter just the duration for a task?
And then let the library calculate the start/end dates? E.g. task A-B-C that depend on each other, A duration 1 day, B duration 2 days + starts Wed, C duration 1 day, D milestone. 
Result: A Tue, B Wed+Thu, C Fri, D Fri/Sat

**Comment by sunnyqijoinhand** [comment]
if you know how to use jQuery , you can solve it like this.
1. give a start date for A , then calculate all tasks start date and end date.
2. draw Gantt chart:
......
 g.Draw();
$('.gmajorheading .gminorheading').empty(); //clear all dates in table cells
.....

**Comment by wivaku** [comment]
Not sure I understand. How do I enter the duration instead of start+end?
And the `empty()` you mention: that just clears the header in the chart, right?

Here's an example of what I would like to do:

```javascript
		g.AddTaskItemObject({
			pID: 1,
			pName: "task A",
			pPlanStart: "", pPlanEnd: "", pStart: "", pEnd: "", 
			pDuration: 8, // <-- possible?
			pClass: "",
			pMile: 0, pComp: 0, 
			pParent: 0, pOpen: 1,
			pRes: "",
			pDepend: "", 
		});

		g.AddTaskItemObject({
			pID: 2,
			pName: "task B",
			pPlanStart: "2019-05-01", pPlanEnd: "", pStart: "", pEnd: "",
			pDuration: 16, // <-- possible?
			pClass: "",
			pMile: 0, pComp: 0, 
			pParent: 0, pOpen: 1,
			pRes: "",
			pDepend: "1", 
		});

		g.AddTaskItemObject({
			pID: 3,
			pName: "task C",
			pPlanStart: "", pPlanEnd: "", pStart: "", pEnd: "", 
			pDuration: 8, // <-- possible?
			pClass: "",
			pMile: 0, pComp: 0, 
			pParent: 0, pOpen: 1,
			pRes: "",
			pDepend: "2", 
		});

		g.AddTaskItemObject({
			pID: 4,
			pName: "task D (milestone)",
			pPlanStart: "", pPlanEnd: "", pStart: "", pEnd: "", 
			pClass: "",
			pMile: 1, pComp: 0, 
			pParent: 0, pOpen: 1,
			pRes: "",
			pDepend: "3", 
		});

		g.Draw();

		$('.gmajorheading .gminorheading').empty(); //clear all dates in table cells
```
and I get:

![image](https://user-images.githubusercontent.com/2084872/56724736-de390e00-674b-11e9-9097-ee68875a0e32.png)

**Comment by mariohmol** [comment]
Task B with Start date  after end date.. did u see this?

**Comment by wivaku** [comment]
Well, that's my point. I would like to enter it as described, and get a result like the 2nd image, and not the 1st.

**Comment by mariohmol** [comment]
I see what u mean... u would like to just set the start date, put a duration and its calculates the end date.. but i dont think we have that yet..

but having start date after end date will always break the chart

**Comment by sunnyqijoinhand** [comment]
give a start date for A , then calculate all tasks start date and end date.

**Comment by wivaku** [comment]
"u would like to just set the start date, put a duration and its calculates the end date": exactly. And: for previous/next related tasks calculate start and end dates based on dependencies.

"i dont think we have that yet": ok, understood. Any plans / alternatives for that? 

Recommendations for any libraries / tools that can read such an "incomplete" data source and produce complete data, with the calculated dates?

**Comment by nasari123** [comment]
you can easly calculate this @wivaku  .. i will give you the algorithm..
set start date of "task A" and calculate end date using duration :
you will  need this function:

function addDays(Startdate, durationInDays) {
  var result = new Date(Startdate);
  result.setDate(result.getDate() + durationInDays);
  return result;
}

so:
endDate = addDays(Startdate, duration);
and startDate of "TASK B " = endDate of "Task A" and so one....

**Comment by wivaku** [comment]
it's the "and so on" that is less trivial than it sounds. 
For each child task I have to recursively check the parents to find the prerequisite with the last completion date.

**Comment by nasari123** [comment]
you can get the id of the parent task from pDep of the child task.
and if you got the id, then you can get start and end date

**Comment by mariohmol** [comment]
thanks to @jonathanlassen and his first #199 on this project =)

**Comment by mariohmol** [comment]
released on v2.2.1
